src/app/[locale]/projects/detail/[id]/components/TreeView.tsx

```diff
@@ -0,0 +1,356 @@
+                    </Link>,
+                    <div className='text-center' key={`${p_id}-projectType`}>
+                        {Capitalize(p.projectType ?? '')}
+                    </div>,
+                    <>{''}</>,
+                    <>{''}</>,
+                    <>{''}</>,
+                    <div className='text-center' key={`${p_id}-state`}>
+                        <OverlayTrigger
+                            overlay={<Tooltip>{`${t('Project State')}: ${Capitalize(p.state ?? '')}`}</Tooltip>}
+                        >
+                            {p.state === 'ACTIVE' ? (
+                                <span className='badge bg-success capsule-left overlay-badge'>{'PS'}</span>
+                            ) : (
+                                <span className='badge bg-secondary capsule-left overlay-badge'>{'PS'}</span>
+                            )}
+                        </OverlayTrigger>
+                        <OverlayTrigger
+                            overlay={
+                                <Tooltip>{`${t('Project Clearing State')}: ${Capitalize(
+                                    p.clearingState ?? ''
+                                )}`}</Tooltip>
+                            }
+                        >
+                            {p.clearingState === 'OPEN' ? (
+                                <span className='badge bg-danger capsule-right overlay-badge'>{'CS'}</span>
+                            ) : p.clearingState === 'IN_PROGRESS' ? (
+                                <span className='badge bg-warning capsule-right overlay-badge'>{'CS'}</span>
+                            ) : (
+                                <span className='badge bg-success capsule-right overlay-badge'>{'CS'}</span>
+                            )}
+                        </OverlayTrigger>
+                    </div>,
+                    <>{''}</>,
+                    <>{''}</>,
+                    <>{''}</>,
+                    <div className='text-center' key={`${p_id}-edit`}>
+                        <OverlayTrigger overlay={<Tooltip>{t('Edit')}</Tooltip>}>
+                            <Link href={`/projects/edit/${p_id}`} className='overlay-trigger'>
+                                <FaPencilAlt className='btn-icon' />
+                            </Link>
+                        </OverlayTrigger>
+                    </div>,
+                ],
+                children: extractLinkedProjectsAndTheirLinkedReleases(
+                    licenseClearingData,
+                    p?.['_embedded']?.['sw360:linkedProjects']
+                ),
+            }
+
+            if (!licenseClearingData['linkedReleases']) {
+                continue
+            }
+
+            for (const l of p['linkedReleases']) {
+                const id = l.release.substring(l.release.lastIndexOf('/') + 1)
+                const res = licenseClearingData['_embedded']['sw360:release'].filter(
+                    (e: any) =>
```

Comment:
This is new code, don't introduce another any variable, proper type it

Reply:
Ok, pass this one for now

Reply:
res variable will be re-calculated everytime it is invoked otherwise. Overall, it will cost a lot of time